### Case description

Luca, Marco and Silvia went out for a dinner. They decided to take a pizza. At the the bill of the restaurant was 40€, they decided to split it flatly, without caring who took what. Luca payed the entire bill. After the pizza, they decided to drink something togheter in a bar. Marco and Silvia took one beer each (5€ per beer) and Silvia payed the entire bill. Luca took a cocktail and Marco decided to pay (10€). They decided to use BillSplitter to split the bill.<br>
Below, you can find the code implementation of the example written above.

### Code implementation

1. Importing the library

In [1]:
import os
import sys
module_path = os.path.abspath(os.path.join('..'))
if module_path not in sys.path:
    sys.path.append(module_path+"\\billsplitter")

from billsplitter import Splitter

2. Creation of the split instance

In [2]:
people_name = ['Luca', 'Marco', 'Silvia']
splitter = Splitter(people_name)

3. Logging the payments

In [3]:
#At the restaurant, Luca payed for everyone
splitter.add_expense(['Luca'], 40)

#At the bar, Silvia payed a beer for Marco and herself
splitter.add_expense(['Silvia'], 10, ['Marco', 'Silvia'])

#At the bar, Marco payed a cocktail for Luca
splitter.add_expense(['Marco'], 10, ['Luca'])

4. Running the calculation

In [4]:
splitter.run_calculation()

5. Getting the net

In [5]:
splitter.net

array([16.66666667, -8.33333333, -8.33333333])

### Resolution

At the end of the night, Marco and Silvia own 8.33€ each to Luca